### PR TITLE
Fix jwtAudience value having {tenantid} instead of common during clie…

### PR DESCRIPTION
**Pull request by jennyf19**
Fix jwtAudience value having {tenantid} instead of common during clie…nt credential w/certificate flow

- add some consts and move log messages to const
- add u tests

[Issue 891](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/891)

### CODE REVIEW

File: src/Microsoft.Identity.Client/Instance/AuthorityEndpointResolutionManager.cs [hunk 1]
```diff
@@ -58,32 +58,52 @@ public AuthorityEndpointResolutionManager(IServiceBundle serviceBundle, bool sho
...
-                edr.Issuer.Replace("{tenant}", tenant));
+                edr.AuthorizationEndpoint.Replace(Constants.Tenant, tenant),
+                edr.TokenEndpoint.Replace(Constants.Tenant, tenant),
+                CreateSelfSignedJwtAudienceEndpointFromIssuer(edr, tenant));
 
             Add(authorityInfo, userPrincipalName, endpoints);
             return endpoints;
         }
 
+        internal string CreateSelfSignedJwtAudienceEndpointFromIssuer(TenantDiscoveryResponse endpoints, string tenant)
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/Instance/AuthorityEndpointResolutionManager.cs]: This method needs explanation in my mind. It needs to say why this is needed

Comment (jennyf19) [reply src/Microsoft.Identity.Client/Instance/AuthorityEndpointResolutionManager.cs]: i agree now that i look at it. how about `ReplaceNonTenantSpecificValueWithCorrectTenant`? 

---
In reply to: [298006468](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1237#discussion_r298006468) [](ancestors = 298006468)

Comment (henrik-me) [reply src/Microsoft.Identity.Client/Instance/AuthorityEndpointResolutionManager.cs]: Like that much better. Perhaps just ReplaceNonTenantSpecificValueWithTenant

---
In reply to: [298222183](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1237#discussion_r298222183) [](ancestors = 298222183,298006468)

**Comment by henrik-me** [comment]
Approving with suggestions to rename as mentioned in the comment

**Comment by jennyf19** [comment]
[integration tests passed here](https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=361616)

**Pull request merged**